src/signax/signatures.py

```diff
@@ -3,36 +3,98 @@
 __all__ = (
     "signature",
     "logsignature",
-    "signature_batch",
     "signature_combine",
     "signature_to_logsignature",
     "multi_signature_combine",
 )
 
 from functools import partial
+from typing import Callable
 
 import jax
 import jax.numpy as jnp
 from jaxtyping import Array, Float
 
+from signax import utils
 from signax.tensor_ops import log, mult, mult_fused_restricted_exp, restricted_exp
-from signax.utils import compress, lyndon_words
 
 
-@partial(jax.jit, static_argnames=["depth", "stream"])
+@partial(jax.jit, static_argnames=("num_chunks", "depth", "stream", "flatten"))
 def signature(
+    path: Float[Array, "path_len dim"] | Float[Array, "batch path_len dim"],
+    depth: int,
+    stream: bool = False,
+    flatten: bool = False,
+    num_chunks: int = 1,
+) -> list[Array] | Array:
+    """
+    Compute the signature of a path. Automatically dispatches to vmap or not based on the shape of `path`.
+
+    Args:
+        path: size (length, dim) or (batch, length, dim)
+        depth: signature is truncated at this depth
+        stream: whether to handle `path` as a stream. Default is False
+        flatten: whether to flatten the output. Default is False
+        num_chunks: number of chunks to use. Default is 1. If > 1, path will be divided into
+        chunks to compute signatures. Then, obtained signatures are combined (using Chen's identity).
+
+    Returns:
+        If `stream` is `True`, this will return a list of `Array` in a form
+            [(path_len - 1, dim), (path_len - 1, dim, dim), (path_len - 1, dim, dim, dim), ...]
+        If `stream` is `False`, this will return a list of `Array` in a form
+            [(dim, ), (dim, dim), (dim, dim, dim), ...]
+        If `flatten` is `True`, this will return a flattened array of shape
+            (dim + dim**2 + ... + dim**depth, )
+        If your path is of shape (batch, path_len, dim), all of the above will have an extra
+        dimension of size `batch` as the first dimension.
+
+    """
+    if num_chunks > 1:
+        sig_fun: Callable[[Array, int, bool, int], Array | list[Array]] = partial(
+            _signature_chunked,
+            num_chunks=num_chunks,
+            depth=depth,
+            stream=stream,
+            flatten=flatten,
+        )
+    else:
+        sig_fun = partial(_signature, depth=depth, stream=stream, flatten=flatten)
+    # this is just to handle shape errors
+    if path.ndim == 2:
+        return sig_fun(path)  # regular case
+    if path.ndim == 3:
+        if flatten:
+            return jax.vmap(sig_fun)(
+                path
+            )  # can use vmap when flattening since it's all arrays
+        return [
+            sig_fun(path[i]) for i in range(path.shape[0])
+        ]  # otherwise, use scan to handle the batch dimension in the list
```

Comment:
I think this is not necessary after we handle `flatten` correctly in the `_signature` function.

Reply:
so i did this because i couldn't figure out how to get the batch dimension as the dimension of the list! It seems like vmap only correctly broadcasts the batch dimension to the arrays, but maybe I fed things in wrong. This is the only thing I did that gave the correct shape for now.

Reply:
this problem does not exist when `flatten=True`, since it's arrays all the way down!